cirq/testing/deprecation.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from contextlib import contextmanager
+
+ALLOW_DEPRECATION_IN_TEST = 'ALLOW_DEPRECATION_IN_TEST'
+
+
+@contextmanager
+def allow_deprecation():
```

Comment:
Would it be possible to make this work like `pytest.raises` where it will _fail_ if no deprecation warning is emitted? Also, it could take a string or regex which is matched against the deprecation warning to ensure that we don't catch things too broadly. These could help ensure that we use `allow_deprecation` in a targeted way, document in code why it's used in specific instances, and remind us to remove it if no longer needed.

Reply:
Sounds good - I was thinking about combining this with the logs assertion, but in the end went with the narrower option. I will create an `assert_deprecated` method instead that combines the two and more importantly makes the version argument mandatory which will make it easier to "flush" a certain version's deprecation backlog.